src/frontend/workers/folderWatcher.worker.ts

```diff
@@ -63,10 +92,16 @@ export class FolderWatcherWorker {
           }
         })
         // .on('change', (path: string) => console.debug(`File ${path} has been changed`))
+        // TODO: on directory change: update location hierarchy list
```

Comment:
I think we could also just store the directory tree in the location object instead of loading everything ansynchronously in the React component. Since we do not remove the children components on closing the outliner anymore, there is no benefit anymore.

Reply:
yeah, I guess. But I feel like this is one of those "if it ain't broke, don't fix" situations

Reply:
We might have to change the ClientLocation a bit anyways. Recently, I realized that one watcher object can watch multiple folders and it would be probably better to have a pool of watcher web workers, so the app performance does not degrade as soon as users add many locations.